test/uk/gov/hmrc/preferencesadminfrontend/controllers/WhitelistControllerSpec.scala

```diff
@@ -0,0 +1,248 @@
+      private val fakeRequestWithForm = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      when(mockMessageConnector.getWhitelist()(any[HeaderCarrier])).thenReturn(
+        Future.successful(
+          HttpResponse(Http.Status.NOT_FOUND)
+        )
+      )
+      private val result = whitelistController.showWhitelistPage()(addToken(fakeRequestWithForm))
+      status(result) shouldBe Status.BAD_GATEWAY
+    }
+
+  }
+
+  "confirmAdd" should {
+
+    "return 303 (Redirect) when a Form ID is successfully added via the message service" in new WhitelistControllerTestCase {
+      private val fakeRequestWithSession = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      private val fakeRequestWithBody: FakeRequest[AnyContentAsJson] = fakeRequestWithSession.withJsonBody(
+        Json.parse(
+          """
+            |{
+            | "formId" : "SA316 2015",
+            | "reasonText" : "some reason text"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.addFormIdToWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
+        Future.successful(
+          HttpResponse(Http.Status.CREATED)
+        )
+      )
+      private val result = whitelistController.confirmAdd()(addToken(fakeRequestWithBody))
+      status(result) shouldBe Status.SEE_OTHER
+    }
+
+    "return 502 (Bad Gateway) when the message service returns any other status" in new WhitelistControllerTestCase {
+      private val fakeRequestWithSession = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      private val fakeRequestWithBody: FakeRequest[AnyContentAsJson] = fakeRequestWithSession.withJsonBody(
+        Json.parse(
+          """
+            |{
+            | "formId" : "SA316 2015",
+            | "reasonText" : "some reason text"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.addFormIdToWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
+        Future.successful(
+          HttpResponse(Http.Status.NOT_FOUND)
+        )
+      )
+      private val result = whitelistController.confirmAdd()(addToken(fakeRequestWithBody))
+      status(result) shouldBe Status.BAD_GATEWAY
+    }
+
+    "return 400 (Bad Request) when the Form ID JSON is not valid" in new WhitelistControllerTestCase {
+      private val fakeRequestWithSession = FakeRequest(routes.WhitelistController.showWhitelistPage()).withSession(User.sessionKey -> "user")
+      val fakeRequestWithBody: FakeRequest[AnyContentAsJson] = fakeRequestWithSession.withJsonBody(
+        Json.parse(
+          """
+            |{
+            | "blah" : "blah"
+            |}
+          """.stripMargin)
+      )
+      when(mockMessageConnector.addFormIdToWhitelist(any[WhitelistEntry])(any[HeaderCarrier])).thenReturn(
```

Comment:
Not sure that mock is required, but I could be wrong

Reply:
Yep, you're right @elabeca. Now removed.